exercises/matrix/src/example/java/Matrix.java

```diff
@@ -0,0 +1,35 @@
+public class Matrix {
+    private int[][] matrix;
+
+    public Matrix(String matrixAsString) {
+        String[] rows = matrixAsString.split("\\n");
+        matrix = new int[rows.length][];
+        for (int i = 0; i < rows.length; i++) {
+            String[] columnValues = rows[i].split(" ");
+            matrix[i] = new int[columnValues.length];
+            for (int j = 0; j < columnValues.length; j++) {
+                matrix[i][j] = Integer.valueOf(columnValues[j]);
```

Comment:
Minor detail, but it's probably ok to use `Integer.parseInt` here instead, which avoid extra boxing/unboxing. See e.g. [this SO post](http://stackoverflow.com/questions/508665/difference-between-parseint-and-valueof-in-java) for some interesting discussion. Not going to hold the PR for this change alone, just noting for interest :)